.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,240 @@
+      
+      - name: effective java version # Yes it actually needs JAVA_HOME=$JAVA_HOME, doesn't use actual env vars (!)
+        if: runner.os == 'Windows'
+        run: java -version
+
+      - name: Query dependencies
+        run: |
+          install.packages('remotes', repos=options('repos')[['repos']][['CRAN']])
+          saveRDS(remotes::dev_package_deps(dependencies = TRUE), ".github/depends.Rds", version = 2)
+          writeLines(sprintf("R-%i.%i", getRversion()$major, getRversion()$minor), ".github/R-version")
+        shell: Rscript {0}
+
+      - name: Cache R packages
+        if: runner.os != 'Windows'
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.R_LIBS_USER }}
+          key: ${{ runner.os }}-${{ hashFiles('.github/R-version') }}-1-${{ hashFiles('.github/depends.Rds') }}${{matrix.config.pkgrepo}}
+          restore-keys: ${{ runner.os }}-${{ hashFiles('.github/R-version') }}-1-
+
+      - name: Install tinytex
+        if: matrix.config.vignettes
+        uses: r-lib/actions/setup-tinytex@v1
+
+      - name: Install extra Tex packages
+        if: matrix.config.vignettes
+        run: |
+          remotes::install_cran('tinytex', repos=options('repos')[['repos']][['RSPM']])
+          library(tinytex)
+          tlmgr_install('datetime') 
+          tlmgr_install('inputenc')
+          tlmgr_install('hyperref')
+          tlmgr_install('url')
+          tlmgr_install('fmtcount')
+          tinytex:::install_yihui_pkgs()
+        shell: Rscript {0}
+
+      - name: Install system dependencies
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt-get install libcurl4-openssl-dev
+          while read -r cmd
+          do
+            eval sudo $cmd
+          done < <(Rscript -e 'writeLines(remotes::system_requirements("${{ matrix.config.os-name }}", "${{ matrix.config.os-version }}"))')
+
+      - name: Install dependencies
+        run: |
+          remotes::install_cran("covr")
+          remotes::install_deps(dependencies = TRUE, repos=options('repos')[['repos']][['RSPM']])
+          remotes::install_cran("rcmdcheck")
+        shell: Rscript {0}
+
+      - name: Session info
+        run: |
+          options(width = 100)
+          pkgs <- installed.packages()[, "Package"]
+          sessionInfo()
+        shell: Rscript {0}
+
+      - name: Check without doc
+        if: ${{!matrix.config.vignettes}}
+        env:
+          _R_CHECK_CRAN_INCOMING_: false
+          FULL_TEST_SUITE: 1
+        run: |-
+            check_args <- if("${{matrix.config.os-name}}" == "windows") {
+              c("--no-manual", "--no-vignettes", "--no-build-vignettes", "--check-subdirs=no", "--no-multiarch")
+            } else {
+              c("--no-manual", "--no-vignettes", "--no-build-vignettes", "--check-subdirs=no")
+            }
+            rcmdcheck::rcmdcheck(args = check_args, build_args=c("--no-build-vignettes"), error_on = "error", check_dir = "check")
```

Comment:
Related to the comment above about `vignette`, should we run `--as-cran` and `error_on = "warning"`?
I guess there might be good reasons why we cannot afford this.

Reply:
This would likely not work on windows and macOS. I'll try changing to vignettes = true on the oldrel and develop ones. I believe this did not work on one of the combinations, will activate it one the ones where it is possible.
I wanted to install tinytex in only one of the builds, as that sometimes takes a while.

Reply:
Update: working builds with `vignettes = true` on all relevant R versions.